extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesCommonHelper.java

```diff
@@ -304,16 +304,16 @@ private static Collection<DecoratorBuildItem> createRbacDecorators(String name,
         }
 
         // Add service account from extensions: use the one provided by the user always
-        String defaultServiceAccount = null;
+        Optional<String> defaultServiceAccount = config.getServiceAccount();
         String defaultServiceAccountNamespace = null;
         for (KubernetesServiceAccountBuildItem sa : serviceAccountsFromExtensions) {
             String saName = defaultIfEmpty(sa.getName(), name);
```

Comment:
I know this line is not changed by this pull request, but why would a `KubernetesServiceAccountBuildItem` contain an empty name? This feels really weird.

Reply:
I mean that it could possibly use Optional instead.

Reply:
We could use the optional, anyhow I prefer not doing this change in this pull request.